modules/core/02-client/keeper/client_test.go

```diff
@@ -440,71 +412,44 @@ func (suite *KeeperTestSuite) TestUpgradeClient() {
 				cs, found := suite.chainA.App.GetIBCKeeper().ClientKeeper.GetClientState(suite.chainA.GetContext(), path.EndpointA.ClientID)
 				suite.Require().True(found)
 
-				upgradedClientProof, _ = suite.chainB.QueryUpgradeProof(upgradetypes.UpgradedClientKey(int64(lastHeight.GetRevisionHeight())), cs.GetLatestHeight().GetRevisionHeight())
-				upgradedConsensusStateProof, _ = suite.chainB.QueryUpgradeProof(upgradetypes.UpgradedConsStateKey(int64(lastHeight.GetRevisionHeight())), cs.GetLatestHeight().GetRevisionHeight())
-			},
-			expPass: false,
-		},
-		{
-			name: "unsuccessful upgrade: upgraded height is not greater than current height",
```

Comment:
I moved this test to the light client module, but can move it back here in the PR where `LatestHeight` is added to the light client module and then we can move the check back to 02-client.

Reply:
https://github.com/cosmos/ibc-go/pull/5827#discussion_r1520067044

I guess it can just be removed entirely now?

Reply:
Sounds good. At the moment this check is performed in 07-tendermint and 08-wasm light client modules. Based on [your comment](https://github.com/cosmos/ibc-go/pull/5827/files#r1520066359), is that ok? Any reason why the check should not be done in 08-wasm?

Reply:
we can have these tests in the lcm modules for 07-tendermint and 08-wasm already, i think! is that fine @crodriguezvega?

Reply:
Yes, correct! The check and the tests are currently in the light client module. Happy to leave it there if that's the right place. :)